third_party/cpython/BUILD

```diff
@@ -0,0 +1,23 @@
+version = "3.12.7"
+release = "20241016"
+
+remote_file(
+    name = "download",
+    url = f"https://github.com/indygreg/python-build-standalone/releases/download/{release}/cpython-{version}+{release}-x86_64_v2-unknown-linux-gnu-pgo+lto-full.tar.zst",
+    hashes = ["98f64550d63fa2f5468b28482b1fed8aef33a70d81f1549b7a8be1a59db41d98"],
+)
+
+genrule(
```

Comment:
Just to clarify, when you talk about "in-repo Python" in the context of this PR do you mean this CPython interpreter that's being built for Python Rules or also another Python interpreter that could be built and used in a repository that uses Please?

I.e., how would this change what interpreters a user would use with Please and where those interpreters are built from?

Reply:
I mean that generally you can use a Python that has been built within the repo, which doesn't have to be a thing on your PATH.

The intention isn't to use the one I added here anywhere else, someone who wanted it would define an appropriate Python in their own repo

Reply:
Got it, thanks! This sounds great.